hedera-node/hedera-smart-contract-service-impl/src/main/java/com/hedera/node/app/service/contract/impl/exec/processors/CustomMessageCallProcessor.java

```diff
@@ -107,7 +109,7 @@ public void start(@NonNull final MessageFrame frame, @NonNull final OperationTra
         if (!alreadyHalted(frame)) {
```

Comment:
I don't get the flow of control here.  If system contract then execute it and return.  (Skip line 104-105 for now) Otherwise if it transfers value then do that _and then continue and check if it is a precompile_.  Why?

Reply:
It's a little tricky because the system contract address range is in the isSystemAccount range.   Also we do not allow transfers to system contract addresses.

Reply:
ok but one of the things I don't get is why, _after_ `doTransferValueOrHalt` there's no `return` - it flows through to L109 where it may or may not be halted and thus may or may not execute L114 (since it isn't a precompile).  If that's what's needed?  Seems convoluted.

Reply:
I'll rework this to try and make it clearer